text/0066-simulation-time.md

```diff
@@ -0,0 +1,119 @@
+- Start Date: (fill in with date at which the RFC is merged, YYYY-MM-DD)
+- RFC PR: [amaranth-lang/rfcs#66](https://github.com/amaranth-lang/rfcs/pull/66)
+- Amaranth Issue: [amaranth-lang/amaranth#0000](https://github.com/amaranth-lang/amaranth/issues/0000)
+
+# Simulation time
+
+## Summary
+[summary]: #summary
+
+Add a type for representing time durations and simulator methods to query the elapsed time.
+
+## Motivation
+[motivation]: #motivation
+
+In [RFC #36](0036-async-testbench-functions.md), there were a plan to introduce a `.time()` method to query the current simulation time, but it got deferred due to the lack of a suitable return type.
+
+Internally simulation time is an integer number of femtoseconds, but exposing that directly is not ergonomic and making it a `float` of seconds sacrifices precision the longer a simulation runs.
+
+Also, to quote @whitequark: Time is not a number.
+
+## Guide- and reference-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+A new type `amaranth.sim.Period` is introduced.
+It is immutable, has no public constructor and is instead constructed through the following classmethods:
```

Comment:
I am not sure if classmethod constructors should have such short names and the time getters the longer one. Usually the construction is more seldom than accessing. So maybe we can have constructors something like `.make_s(duration)`, `.make_ms(duration)`... and the access functions as `.get_s()`, `.get_ms()`,...

Reply:
Not in this case though. You will almost never use accessors. You should keep that in mind when evaluating the API design.